mailpoet/views/segments.html

```diff
@@ -152,12 +152,22 @@
     'equals': __('equals'),
     'notEquals': __('not equals'),
     'contains': __('contains'),
+    'notContains': _x('does not contain', 'Dynamic segment creation: operator used with subscriber fields. Should make sense in a sentence like "subscriber email does not contain hotmail.com"'),
     'value': __('value'),
     'selectValue': __('Select value'),
     'checked': __('checked'),
     'unchecked': __('unchecked'),
     'unknown': __('unknown'),
     'notUnknown': __('not unknown'),
+    'is': _x('is', 'Dynamic segment creation: operator used with subscriber fields. Should make sense in a sentence like "subscriber first name is Bill"'),
+    'isNot': _x('is not', 'Dynamic segment creation: operator used with subscriber fields. Should make sense in a sentence like "subscriber first name is not Bill"'),
+    'startsWith': _x('starts with', 'Dynamic segment creation: operator used with subscriber fields. Should make sense in a sentence like "subscriber email starts with bill"'),
+    'notStartsWith': _x('does not start with', 'Dynamic segment creation: operator used with subscriber fields. Should make sense in a sentence like "subscriber email does not start with bill"'),
+    'endsWith': _x('ends with', 'Dynamic segment creation: operator used with subscriber fields. Should make sense in a sentence like "subscriber email ends with yahoo.com"'),
+    'notEndsWith': _x('does not end with', 'Dynamic segment creation: operator used with subscriber fields. Should make sense in a sentence like "subscriber email does not end with yahoo.com"'),
+    'firstName': _x('first name', 'Dynamic segment creation: option in the segment select. Should make sense in a sentence like "subscriber first name does not contain Q"'),
+    'lastName': _x('last name', 'Dynamic segment creation: option in the segment select. Should make sense in a sentence like "subscriber last name does not contain Q"'),
+    'email': _x('email', 'Dynamic segment creation: option in the segment select. Should make sense in a sentence like "subscriber email does not contain Q"'),
```

Comment:
Context is not meant to be used like that. Each context creates a new string, and it serves to differentiate strings, such as `trigger` (a noun) vs. `trigger` (a verb). If you want to add more "context" in terms of documentation, then a `// translators: ...` comment serves that purpose (it's included with extracted strings and displayed in translation UIs).

Reply:
@johnolek

> I feel like this should be clear from other conversations, but just to confirm, are we not using translators: ... comments in our views/templates/twig files because they wouldn't be included in the output the way we process them, is that right? And it is possible to include those comments in typescript files as long as we're using @wordpress/i18n functions directly instead of the MailPoet.I18n wrapper?

With `@wordpress/i18n` the comments should be extracted, but we need to fix our configuration to do that first. For the views and Twig templates, I think it's not worth fixing, we could consider them deprecated.

> For many of these strings I should be able to re-use existing translations and just make sure they're lowercase. Are translations case sensitive? Would __('word') and __('Word') result in two translations?

I'm not sure, but I think they are case sensitive.

> The only translation that stood out to me as a potential issue was email since it could be both a verb and a noun, but I think we only ever use it as a noun in the codebase right now so there shouldn't be a need to distinguish. Does that sound right to you?

Probably no need trying to fix that now, there will be some tickets trying to address these issues.